File: dart/longest_common_prefix.dart
class Solution {
  String longestCommonPrefix(List<String> strs) {
    if (strs.isEmpty) {
      return "";
    }
    if (strs.length == 1) {
      return strs[0];
    }
    String referenceString = strs[0];

    for (int i = 0; i < referenceString.length; i++) {
      String charToCompare = referenceString[i];

      for (int j = 1; j < strs.length; j++) {
        String otherString = strs[j];

        if (i >= otherString.length || otherString[i] != charToCompare) {
          return referenceString.substring(0, i);
        }
      }
    }
    return referenceString;
  }
}

File: python/longest_common_prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        if not strs:
            return ""
        if len(strs) == 1:
            return strs[0]
        referenceString = strs[0]

        for i in range(len(referenceString)):
            charToCompare = referenceString[i]
            for j in range(1, len(strs)):
                otherString = strs[j]
                if i >= len(otherString) or otherString[i] != charToCompare:
                    return referenceString[0:i]
        return referenceString
